src/Morphir/Type/Infer.elm

```diff
@@ -68,9 +69,10 @@ inferModuleDefinition refs moduleName moduleDef =
                     _ =
                         Debug.log
                             (String.concat [ "Inferring types for ", moduleName |> Path.toString Name.toTitleCase ".", ".", valueName |> Name.toCamelCase, " of size" ])
-                            (valueDef.value.body |> Value.countValueNodes)
+                            (valueDef.value.value.body |> Value.countValueNodes)
                 in
-                inferValueDefinition refs valueDef.value
+                inferValueDefinition refs valueDef.value.value
+                    |> Result.map (Documented "")
```

Comment:
We should retain the documentation here. You can use this function to do that: https://github.com/finos/morphir-elm/blob/8ae877f1d937c1f6042f33970698312bd1d47c5f/src/Morphir/IR/Documented.elm#L32-L34